### DISCUSSION TITLE
Setting site Logo (vs. Thumbnail) on Modern Sites (no Group)

### DISCUSSION THREAD

**Discussion by skantere** [pnp/powershell#797] [Q&A]
Hi, 

I'm looking for the way to change both logo and thumbnail for the root web and all subwebs on any site; many of my sites are Modern Sites without O365 Group (STS#3). I use a recursive function and for each web do the following:
`Set-PnpWeb -SiteLogoUrl $SiteLogoUrl` 
And it works like a charm (for newly created sites)! Both logo and thumbnail (seem to) get changed. 

But as soon as I change the site logo on STS#3-based site in UI, it "sticks". Any further attempts of `Set-PnpWeb -SiteLogoUrl` set just the thumbnail - the logo remains unaffected. The only way to make it configurable through `Set-PnpWeb` again is to completely remove it in UI.

So I notice this in the docs:
_If you want to set the logo to a modern team site, use Set-PnPSite -LogoFilePath._
Aha! I run this
`Set-PnpSite -LogoFilePath $LogoFilePath`
but sadly get an exception "Set-PnpSite : Not an Office365 group enabled site."

So I'm completely stuck... Am I missing something?

**Comment by skantere** [comment]
Actually, I have just tried
`Set-PnpSite -LogoFilePath $LogoFilePath`
on a regular Modern Site (with group), and this time I don't get errors, but logo still fails to be overwritten.

And I wonder what would be the command to remote the logo? It's possible in UI, shouldn't it also be possible in PS?

**Comment by veronicageek** [comment]
I've transferred this as a discussion, unless you have a command that's meant to work and it doesn't with the new version.

**Reply by skantere** [reply]
Hallo Veronique,

I do have a command that's supposed to work, but it does not. It's the second line in this short script:
`Connect-PnPOnline -Url "https://mycompany.sharepoint.com/sites/mysite" -Interactive`
`Set-PnpSite -LogoFilePath "c:\dummy.png"`

This is supposed to set the logo on modern team sites. I know of two types of modern team sites: with associated O365 group and without one:
1. When I try the script on a modern team site WITH O365 group, I get no error message as if everything has worked properly, however the logo does NOT get changed. The thumbnail DOES get changed.
2. When I try the script on a modern team site WITHOUT O365 group, I get an exception: _"Set-PnpSite : Not an Office365 group enabled site."_ (So what? It's still a modern team site!)

Please note however that the problem was not introduced with the new version of PnP PowerShell. The general problem with updating logo was likely introduced with the new SharePoint site header options as described here: 
https://ganeshsanapblogs.wordpress.com/2020/12/08/new-site-header-options-are-coming-to-sharepoint-online/
There are now two "flavors" of logo: "logo" and "thumbnail", and only one switch -LogoFilePath on Set-PnPSite command! Until the new switch is introduced, I would expect -LogoFilePath to update just the logo, or both, but for God's sake not the way it is now, which is completely counter-intuitive.

As for the problem of discriminating modern team sites without a group, it also existed in SharePointPnPPowerShellOnline, at least in the most recent versions. I believe it's a bug unless I fundamentally miss something.
